### DISCUSSION TITLE
Feedbacks regarding the new TUI

### DISCUSSION THREAD

**Discussion by magne4000** [vercel/turborepo#8381] [Ideas]
### Goals

The goal here is to list some improvements that could be made to the newly released TUI

### Background

There was a need to [have better feedback](https://github.com/vercel/turbo/issues/4042) with the streamed version of the UI, which I think is partly fixed now with theb TUI, but new minor inconvenience are now appearing.

### TUI could follow active tasks
When running a lot of tasks (>50), the active elements chosen by default seems random currently. IMO it should keep focus on the first or last actively running task in the list, and jump to the next one on finish. Any user input should probably then fallback to sticking to what the user chose.

### TUI summary is too verbose
When finishing, TUI currently generates a grouped overview of all tasks with their log.
When having >50 tasks, it becomes noise. Even when `--output-logs=errors-only`, groups are printed, and finding the error is more difficult than it should be.

First, I would suggest that if a group has no output, just do not print the group (or at least give an option in CLI to opt-in this behavior).

Then, I would like an equivalent to `--output-logs`, but that only impact the grouped summary, not the TUI view. That way we can have details about running tasks, and only print a summary of failed ones.

**Comment by solsson** [comment]
> When having >50 tasks, it becomes noise. Even when --output-logs=errors-only, groups are printed, and finding the error is more difficult than it should be.

I'm definitely +1 on this. While in interactive mode it's reasonably doable to spot which tasks that failed and sometimes you have time to keypress your way to those logs. But after termination it's very hard when there's many tasks. Turbo 1.x would at least list failed tasks (comma separated list of packagename#taskname) at the end so you could look for those logs.

Was `--continue` in scope for TUI? We tend to rely on it in QA in order to minimize reruns when there's multiple issues. I've tried with and without `--log-order=stream` and `--output-logs=errors-only` but none of the permutations give me a list of failed tasks at the end.

**Reply by solsson** [reply]
> Turbo 1.x would at least list failed tasks (comma separated list of packagename#taskname) at the end so you could look for those logs.

`--summary=true` solves this. Sorry I didn't find that flag last time.

**Reply by amritk** [reply]
Could you please link the documentation for summary=true? I could not find it, nor get it to work

**Reply by felipedeboni** [reply]
@solsson `--summary` does not seem to exist in turbo 2.0.6, there's `summarize`, but it automatically closes and create a log in json format. Not as easier as `tui` to find the information you're looking for, at least if the monorepo has lots of packages like ours.

**Reply by solsson** [reply]
Sorry for the typo. It sure is --summarize=true and you have to use jq on the path that turbo prints. For example `jq -r '.tasks | .[] | select(.execution.exitCode != 0) | (.taskId + "\n  " + .execution.error)' [path]`

**Comment by pieterjandebruyne** [comment]
I also miss the ability to copy log output from the terminal (macOs)

**Reply by ftzi** [reply]
You can nowadays press `c` to copy selected text. But cmd+f to find text is useless as it only searches in the current visible output, it doesn't look before or after.

**Comment by amritk** [comment]
Yes! how do we get an errors summary after OR how do we just keep the TUI running so we can scroll to the package that has errors? Currently with --continue it keeps going but still closes at the end.

**Reply by solsson** [reply]
Does `--summary=true` meet your need?

**Reply by amritk** [reply]
You mean [summarize](https://turbo.build/repo/docs/reference/run#--summarize)? It generates a json file. I was looking to simply show the error after the command fails.

**Reply by solsson** [reply]
Yes. It's not ideal but at least I get the information needed using jq.

**Reply by anthonyshew** [reply]
If you want the TUI to keep running, that means you still want your `turbo` invocation to be running. [`turbo watch`](https://turbo.build/repo/docs/reference/watch) will do this, allowing you to continue working with your repository's tasks still running.

**Comment by Jac0xb** [comment]
Would be nice to hide, sort (give priority), group tasks (packages / services) in the TUI.

**Reply by zdaar** [reply]
Using it with next + wrangler its very annoying that you cant assign a static order to your tasks, which means you always have to scan the x amount of tasks you have to find where to go. Glad I'm not the only one and I am surprised this is not talked about more.  something like : 

```json
  "dev": {
    "persistent": true,
    "ui": { "priority": 1 } 
  }
  ```
